pkg/apis/cartographer/v1alpha1/workload_helpers.go

```diff
@@ -332,22 +332,29 @@ func (w *WorkloadSpec) MergeGit(git GitSource) {
 	w.Source = &Source{
 		Git: &git,
 	}
-	if stash != nil && stash.Git != nil {
-		if w.Source.Git.URL == "" {
-			w.Source.Git.URL = stash.Git.URL
-		}
-		if w.Source.Git.Ref.Branch == "" {
-			w.Source.Git.Ref.Branch = stash.Git.Ref.Branch
+	if stash != nil {
+		w.Source.Subpath = stash.Subpath
+		if stash.Git != nil {
+			if w.Source.Git.URL == "" {
+				w.Source.Git.URL = stash.Git.URL
+			}
+			if w.Source.Git.Ref.Branch == "" {
+				w.Source.Git.Ref.Branch = stash.Git.Ref.Branch
+			}
 		}
 	}
 }
 
 func (w *WorkloadSpec) MergeSourceImage(image string) {
+	stash := w.Source
 	w.ResetSource()
 
 	w.Source = &Source{
 		Image: image,
 	}
+	if stash != nil {
+		w.Source.Subpath = stash.Subpath
```

Comment:
in this line, I also tried with 
```
w.MergeSubPath(stash.Subpath)
```

but decided not to go with this approach since this latest mentioned method is checking `if w.Source != nil (...)` which is a validation that's already being done in the calling function. 

But it's also an option